uthdev/mock-shop
# #171184290 Build User Signup endpoint

#### Description of Tasks to be completed?
- setup controller for  user registration
- setup route for  user registration
- setup unit test for user registration route

 #### How should this be manually tested?
- cd into the cloned directory
- make sure database connection is already set up
- run `npm install`
- ADD **SECRET**= Just another secret to **.env**
- To run database migration  `npm run migration`
- To test run `npm run test`
- run `npm run start:dev`
- To test with postman use `http://localhost:3000/api/v1/users/signup` with appropriate parameters

 #### Any background context you want to provide?
- Sample request:
```json
{
        "firstName": "firstname",
	"lastName": "lastname",
	"email": "emails@gmail.com",
	"password": "password"
}
```

- Sample response :
 ```json
{
    "status": "success",
    "data": {
        "id": 7,
        "firstName": "firstname", 
	    "lastName": "lastname",
        "email": "emails@gmail.com",
        "updatedAt": "2019-08-22T13:07:31.238Z",
        "createdAt": "2019-08-22T13:07:31.238Z",
        "token": "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJlbWFpbCI6ImVtYWlsZXNzZWRAZ21haWwuY29tIiwidXNlcl9pZCI6NywiaXNfYWRtaW4iOmZhbHNlLCJmaXJzdF9uYW1lIjoiRmlyc3RuYW1lIiwibGFzdF9uYW1lIjoiTGFzdG5hbWUiLCJpYXQiOjE1NjY0NzkyNTEsImV4cCI6MTU2NjczODQ1MX0.3-m7JqTXuuuMZneqUDRfyCd1UCgZtFRa2Z22XIE_ldg"
    }
}
```
 #### What are the relevant pivotal tracker stories?
[#171184290](https://www.pivotaltracker.com/story/show/171184290)
#### Screenshots (if appropriate)
None
 #### Questions:
None
